lib/utils/ensureExists.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+const fse = require('fs-extra');
+const fs = require('fs');
+const path = require('path');
+
+module.exports = async (cacheDir, fileName, generate) => {
```

Comment:
I will probably keep signature of this module to `filename, generate`, where we expect `filename` to be absolute path to filename

(return of `cacheDir` seems specific to `ensureArtifact`, and I think it shouldn't leak here)